packages/react-relay/modern/ReactRelayRefetchContainer.js

```diff
@@ -260,6 +260,15 @@ function createContainerWithFragments<TBase: ReactClass<*>>(
       };
     };
 
+    getChildContext(): Object {
+      return {
+        relay: {
+          environment: this.context.relay.environment,
+          variables: this._localVariables || this.context.relay.variables,
+        },
```

Comment:
We have an optimization in all of the containers that they ignore context changes whenever the `context.relay` is `===` to its previous value and forcibly update whenever it is `!==` to the previous value. To correspond with this `QueryRenderer` returns a new `relay` value whenever the environment or variables actually changes.

Let's apply similar logic here:

```
getChildContext() {
  return {relay: this._relayContext};
}
```

And then set `this._relayContext` as follows:
- initialize in the constructor
- overwrite with a new variable whenever a `refetch` completes succesfully.